contracts/deployment_scripts/layer1/001_deploy_management_contract.ts

```diff
@@ -0,0 +1,22 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
```

Comment:
Not seeing the value of using the deploy package. Can you ping across some examples ? 
I would prefer to avoid future-proofing, better to address the needs as we have them.

Reply:
At the stage of only deploying to L1 there isn't much of a value, apart from the checks if contracts exist and not deploying multiple times, but when deploying to L1 and L2 and requiring to maintain connection and deployed state across two networks it makes it easier due to being able to define companion networks and then calling functions like

```
layer1.execute(...
layer2.execute(...
```

The HRE is not really good by itself for multiple networks.